### ISSUE TITLE
False positive for some URLs on Mac

### ISSUE THREAD

**Issue by alexgarciab** [rvalitov/backlink-checker#1]
Hi there, I have been testing the script and it works good for most URLs. However, there are some URLs that show "link NOT found" but when you check it manually, the link is there, so that means it's a false positive. I activated the verbose mode, but still, not showing much.

I was wondering if there is a way to check the URL Response Code on those URLs that did not match. I guess they implement some kind of block to "wget" or similar.

**Comment by rvalitov** [comment]
Hi! If the website blocks wget, then you can simply put the desired user agent string in the start options of the script as specified in the README. If this does not help, then please provide more details and specify the web page you're testing and the exact arguments that you pass to this script.

**Comment by alexgarciab** [comment]
Hello! Yes, I tried the "User Agent" option, and it worked for some websites, but not for all of them where I was having problems. I will provide you an example of a website where the script is not working, localizing existing links (you can try to search for spyzie.com):

https://www.madridiario.es/noticia/469797/perdiste-tu-celular-aqui-te-ensenamos-como-puedes-localizarlo.html

**Comment by alexgarciab** [comment]
Also, I have detected a bug on the checker. The last link of the URL list, provided by parameter, does not get checked at any time.

**Comment by rvalitov** [comment]
I checked your example, it works:
```console
user@server:/home/ram# bash backlink-checker.sh -input test.txt -link spyzie.com
✔ https://www.madridiario.es/noticia/469797/perdiste-tu-celular-aqui-te-ensenamos-como-puedes-localizarlo.html link found
```

Regarding the last line, what editor do you use? I use `nano` and have no problems with last line. As a workaround I can suggest to add a blank line at the end of the file. But again, I can't reproduce your problem.

**Comment by alexgarciab** [comment]
That "domain" in particular, is not working for me. It is weird that it works for you. Is there a way to activate debug mode (response headers mainly) to see what is causing that check to fail?

In regards to the editor, I use Sublime Text 3. Adding a blank line at the end, solved the problem, but ideally, it should work without having to do that hack.

**Comment by rvalitov** [comment]
May be it's a problem of line endings. What line endings do you use? Linux-style or Windows-style? And what OS do you use? In my case I tested with latest stable Debian, and everything was fine.

**Comment by alexgarciab** [comment]
I use Mac OS Catalina. I don't use any special kind of line ending. Just open up Sublime, copy the urls (one per line), save the file as .txt and check it with the terminal command.

**Comment by rvalitov** [comment]
I checked on Macbook with iOS Mojave and I confirm that it does not find the URL. I will try to fix that.

**Comment by rvalitov** [comment]
@alexgarciab after investigation the problem was not in my code, the problem was in iOS binaries. The `grep` was outdated and didn't work properly. I updated the README how to properly install and configure the prerequisites in iOS. Please, follow it, and it should solve the problem that the link is not found.

I tried different editors, but couldn't reproduce the problem that the last line of the links file is ignored. If this problem still happens for you, please, open a new issue and attach your file with links there, so that I can test it and check the encoding, line endings, etc..
